ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -98,86 +102,109 @@ func (r *RayServiceReconciler) Reconcile(ctx context.Context, request ctrl.Reque
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	var rayClusterInstance *rayv1alpha1.RayCluster
-	if rayClusterInstance, err = r.getOrCreateRayClusterInstance(ctx, rayServiceInstance); err != nil {
-		err = r.updateState(ctx, rayServiceInstance, rayv1alpha1.FailToGetOrCreateRayCluster, err)
+	// Reconcile RayCluster, check if we need to create a new RayCluster.
+	var activeRayClusterInstance *rayv1alpha1.RayCluster
+	var pendingRayClusterInstance *rayv1alpha1.RayCluster
+	if activeRayClusterInstance, pendingRayClusterInstance, err = r.reconcileRayCluster(ctx, rayServiceInstance); err != nil {
+		err = r.updateState(ctx, rayServiceInstance, rayv1alpha1.FailedToGetOrCreateRayCluster, err)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	log.Info("Updated RayCluster")
+	// Reconcile Serve Configs.
+	r.cleanUpServeConfigMaps(rayServiceInstance)
```

Comment:
Can you say again what these config maps are?
How are they used?

Reply:
I just understood -- we're cleaning up cached state.
Could you rename this method `r.cleanUpServeConfigCache` or something?
(I see cleanupConfigMap and assume we're reconciling config map objects.)